Example/lightbulb

AceMouty opened the pull request:
A demo using a Signal and Show component to conditionally render elements
to the UI.

Features:

- createSignal
- Read / Update signal
- handle click event
- Show component

![lightbulb-result](https://user-images.githubusercontent.com/45374681/174958879-eda56519-2139-49ef-86a2-9efd980f5c64.gif)

AceMouty committed:
feat: created demo using signal and show component

basic demo to track state and conditionally render elements to the UI

AceMouty committed:
chore: updated README with demo gif

AceMouty committed:
chore: fixed typos in README

edemaine commented on examples/basic/lightbulb/src/App.tsx:
```diff
@@ -0,0 +1,22 @@
+      <h1 class={styles.header}>Click the light to turn it on / off</h1>
+      <div class={styles.bulbContainer} onClick={() => setLightOn((lightOnValue) => !lightOnValue)}>
+        <Show when={lightOn()} fallback={<img src={bulbOff} />}>
+          <img src={bulbOn} />
```
This would be more efficient without `<Show>`:

```jsx
        <img src={lightOn() ? bulbOn : bulbOff} />
```

This avoids DOM creation/deletion/mounting. But sadly no longer illustrates `<Show>`. 😅

atk replied:
I think it would be nicer if the lightbulb was SVG and an additional <g>roup of paths to light it up was put inside <Show>.

edemaine replied:
Oh yeah, that'd be the best of both worlds -- then we could illustrate `<Show>` again.

AceMouty replied:
So, there is a lot more examples to come where we can better demo `<Show>` and Im all for doing this here

```jsx
        <img src={lightOn() ? bulbOn : bulbOff} />
```

But I'm hesitant to do the SVG route. The idea for this demo is just something really basic where a user can come to as a reference to figure out how to track state in Solid while cutting out as much noise as possible in the code backing the example.

AceMouty committed:
cleanup: simplified demo

simplified conditonal logic for the src of the img tag
moved logic to be a inline ternary for the img src

mosheduminer commented:
I'm ready to merge this, I'm just wondering why the light bulb moves when it's clicked? I suppose the image dimensions are slightly different?

AceMouty commented:
Correct, image dimensions are slightly different. I had a image that has the light both on & off and just cropped it into 2 different frames 😅
